### ISSUE TITLE
Is anyone running Sisimai on JRuby?

### ISSUE THREAD

**Issue by azumakuniyuki** [sisimai/rb-sisimai#267]
- Sisimai does not work JRuby 9.2 or later
- Test code fails on JRuby 9.1
- **We are considering dropping JRuby support if there are no or very few users using Sisimai with JRuby.**

**Comment by azumakuniyuki** [comment]
https://gist.github.com/hiroyuki-sato/6ef40245874d4c847a95ef99886e4fa7

**Comment by hiroyuki-sato** [comment]
Minimal reproducing code.

MRI -> No Exception.
JRuby(9.4.3.0) -> Exception

```ruby
require 'sisimai/datetime'
require 'sisimai/time'

datestring = 'Tue, 28 Apr 2009 09:52:45'
begin
  t = Sisimai::Time.strptime(datestring, '%a, %d %b %Y')
  #t.to_time.to_i
  t.to_time
rescue
  print $!.backtrace.join("\n")
end
```

[sisimai/datetime.rb](https://github.com/sisimai/rb-sisimai/blob/5-stable/lib/sisimai/datetime.rb)
sisimai/time.rb

```ruby
require 'date'
module Sisimai
  # Sisimai::Time is a child class of Date for Sisimai::Fact
  class Time < ::DateTime
    def to_json(*)
      return self.to_time.to_i
    end
  end
end
```


```
org.jruby.dist/org.jruby.ext.date.RubyDateTime$INVOKER$i$0$0$to_time.call(RubyDateTime$INVOKER$i$0$0$to_time.gen)
org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:456)
org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:195)
test.invokeOther3:to_time(test.rb:8)
test.RUBY$script(test.rb:8)
test.run(test.rb)
java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
org.jruby.dist/org.jruby.ir.Compiler$1.load(Compiler.java:114)
org.jruby.dist/org.jruby.Ruby.runScript(Ruby.java:1276)
org.jruby.dist/org.jruby.Ruby.runNormally(Ruby.java:1193)
org.jruby.dist/org.jruby.Ruby.runNormally(Ruby.java:1175)
org.jruby.dist/org.jruby.Ruby.runNormally(Ruby.java:1211)
org.jruby.dist/org.jruby.Ruby.runFromMain(Ruby.java:989)
org.jruby.dist/org.jruby.Main.doRunFromMain(Main.java:398)
org.jruby.dist/org.jruby.Main.internalRun(Main.java:282)
org.jruby.dist/org.jruby.Main.run(Main.java:227)
org.jruby.dist/org.jruby.Main.main(Main.java:199)
```

Reason

```
class org.jruby.ext.date.RubyDate cannot be cast to class org.jruby.ext.date.RubyDateTime (org.jruby.ext.date.RubyDate and org.jruby.ext.date.RubyDateTime are in module org.jruby.dist of loader 'app')org.jruby.dist/org.jruby.ext.date.RubyDateTime$INVOKER$i$0$0$to_time.call(RubyDateTime$INVOKER$i$0$0$to_time.gen)
```

**Comment by azumakuniyuki** [comment]
@hiroyuki-sato Thank you for the advise. I will try to fix the compatibility issue with JRuby within a few weeks.

**Comment by hiroyuki-sato** [comment]
I asked about this behavior. https://github.com/jruby/jruby/issues/8139

**Comment by hiroyuki-sato** [comment]
In this part, the following code may fix the issue. It is necessary more rework.

```ruby
module Sisimai
  # Sisimai::Time is a child class of Date for Sisimai::Fact
  class Time
    class << self
      def strptime(str,format)
        t = ::DateTime.strptime(str,format)
        self.new(t)
      end
      def parse(t)
        t = ::DateTime.parse(t)
        self.new(t)
      end
    end
    def initialize(dtime)
      @dtime = dtime
    end
    def to_time
      return @dtime.to_time
    end
    def to_json(*)
      return @dtime.to_time.to_i
    end
  end
end
```

**Comment by hiroyuki-sato** [comment]
JFYI @azumakuniyuki 

This issue will be fixed in the next JRuby release(9.4.7.0).
The code has been fixed.
https://github.com/jruby/jruby/pull/8144

**Comment by azumakuniyuki** [comment]
This issue resolved at v5.0.2
